framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -12,24 +12,59 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { BasePlugin, PluginInitContext } from 'lisk-sdk';
+import { BasePlugin, PluginInitContext, Transaction, apiClient } from 'lisk-sdk';
+import { CCM_BASED_CCU_FREQUENCY, LIVENESS_BASED_CCU_FREQUENCY } from './constants';
+import { configSchema } from './schemas';
+import { ChainConnectorPluginConfig } from './types';
 
-export class ChainConnectorPlugin extends BasePlugin {
+interface CCUFrequencyConfig {
+	ccm: number;
+	liveness: number;
+}
+
+type SentCCUs = Transaction[];
+
+export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig> {
 	public name = 'chainConnector';
+	public configSchema = configSchema;
+	private _lastCertifiedHeight!: number;
+	private _ccuFrequency: CCUFrequencyConfig = {
+		ccm: CCM_BASED_CCU_FREQUENCY,
+		liveness: LIVENESS_BASED_CCU_FREQUENCY,
+	};
+	private _mainchainAPIClient!: apiClient.APIClient;
```

Comment:
what's the reason for null assertion for `_mainchainAPIClient` ?

Reply:
its not assigned in constructor but in the load() so it throws lint error if I don't assign null assertion